lib/repeatable/expression/base.rb

```diff
@@ -24,6 +24,29 @@ def to_h
         )
       end
 
+      def union(other)
+        if other.is_a?(Union)
```

Comment:
Can you help me understand the desire for these sorts of special cases to "reduce redundant nesting?"  Was there a specific use-case or problem scenario that made you feel like this was necessary?

I understand the general desire to reduce redundancy, but I'm not sure that the added complexity and cognitive load of this code is worth it. I lean toward letting the redundancy happen in favor of keeping this implementation as simple as possible and letting the user work out the most efficient way to build up sets.